subsys/bluetooth/host/mesh/friend.c

```diff
@@ -1190,7 +1192,7 @@ static bool friend_lpn_matches(struct bt_mesh_friend *frnd, u16_t net_idx,
 	}
 
 	if (BT_MESH_ADDR_IS_UNICAST(addr)) {
-		if (addr == frnd->lpn) {
+		if (addr >= frnd->lpn && addr < (frnd->lpn + frnd->num_elem)) {
```

Comment:
Perhaps it would be a good idea to have a helper function to resolve valid friend addresses like above.

Reply:
Perhaps. Currently there are two places needing it.

Reply:
Fixed now